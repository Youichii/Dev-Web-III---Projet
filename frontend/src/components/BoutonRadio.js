import PropTypes from 'prop-types'

const BoutonRadio = ({ className_div, id_div, name, value, form, text, checked, onClick }) => {
    return (
        <div className={className_div} id={id_div}>
            <input type="radio" name={name} value={value} id={form} checked={checked} onClick={onClick} />
            <label for={form} className="label-info">{text}</label>
        </div>
    )

}

/*BoutonRadio.defaultProps = {
    checked: 'no'
}*/

BoutonRadio.propTypes = {
    className : PropTypes.string,
    id_div : PropTypes.string,
    id_elem : PropTypes.string,
    name : PropTypes.string,
    value : PropTypes.string,
    onClick : PropTypes.func,
}

export default BoutonRadio