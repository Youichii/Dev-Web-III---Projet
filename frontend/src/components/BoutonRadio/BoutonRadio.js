import PropTypes from 'prop-types'

const BoutonRadio = ({ className_div, id_div, name, value, form, text, isChecked, onClick }) => {
    return (
        <div data-testid="div_bouton_radio" className={className_div} id={id_div}>
            <input data-testid="input_bouton_radio" type="radio" name={name} value={value} id={form} defaultChecked onClick={onClick} readOnly/>
            <label data-testid="label_bouton_radio" htmlFor={form} className="label-info">{text}</label>
        </div>
    )
}

/*BoutonRadio.defaultProps = {
    checked: 'no'
}*/

BoutonRadio.defaultProps = {
    isChecked : "checked"
}

BoutonRadio.propTypes = {
    className : PropTypes.string,
    id_div : PropTypes.string,
    id_elem : PropTypes.string,
    name : PropTypes.string,
    value : PropTypes.string,
    onClick : PropTypes.func,
}

export default BoutonRadio
