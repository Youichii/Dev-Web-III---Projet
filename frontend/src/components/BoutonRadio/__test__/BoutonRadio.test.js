import React from 'react';
import ReactDOM from 'react-dom';
import BoutonRadio from '../BoutonRadio';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer  from "react-test-renderer";

afterEach(cleanup);
it("renders without crashing BoutonRadio", () => {
    const div = document.createElement("div");
    ReactDOM.render(<BoutonRadio></BoutonRadio>, div)
})

it("get correctly value = 'homme' BoutonRadio", () => {
    const {getByTestId} = render(<BoutonRadio className_div="i_mistercash" id_div="radio_mistercash" name="myradio2" value="homme" form="mistercash" text="Mistercash" checked="yes"></BoutonRadio>);
    expect(getByTestId('input_bouton_radio')).toHaveAttribute('value', 'homme');
})

it("render text correctly BoutonRadio", () => {
    const {getByTestId} = render(<BoutonRadio text="Mistercash" ></BoutonRadio>);
    expect(getByTestId('label_bouton_radio')).toHaveTextContent("Mistercash");
})

it("render checked correctly BoutonRadio", () => {
    const {getByTestId} = render(<BoutonRadio checked="yes"></BoutonRadio>);
    expect(getByTestId('input_bouton_radio')).toHaveProperty('checked', true);
})

it("matches snapshot 1 BoutonRadio", () => {
    const tree = renderer.create((<BoutonRadio></BoutonRadio>)).toJSON(); 
    expect(tree).toMatchSnapshot();
})
