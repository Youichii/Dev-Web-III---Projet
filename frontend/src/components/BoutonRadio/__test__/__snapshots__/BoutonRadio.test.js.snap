// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 BoutonRadio 1`] = `
<div
  data-testid="div_bouton_radio"
>
  <input
    checked="yes"
    data-testid="input_bouton_radio"
    readOnly={true}
    type="radio"
  />
  <label
    className="label-info"
    data-testid="label_bouton_radio"
  />
</div>
`;
